src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,801 @@
+        {
+            final ResetFW reset = resetRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                                         .routeId(routeId)
+                                         .streamId(initialId)
+                                         .trace(traceId)
+                                         .build();
+
+            network.accept(reset.typeId(), reset.buffer(), reset.offset(), reset.sizeof());
+        }
+
+        private void doSocksHandshakeReply(
+            SocksAuthenticationMethod method)
+        {
+            SocksHandshakeReplyFW handshakeReply = handshakeReplyRW.wrap(writeBuffer,
+                                                                         DataFW.FIELD_OFFSET_PAYLOAD,
+                                                                         writeBuffer.capacity())
+                                                                   .version(5)
+                                                                   .method(method.value())
+                                                                   .build();
+
+            doNetworkData(handshakeReply.buffer(), handshakeReply.offset(), handshakeReply.sizeof());
+        }
+
+        private void doSocksCommandReply(
+            String address,
+            int port)
+        {
+            byte[] ipv4Address = lookupName(address).getAddress();
+            SocksCommandReplyFW socksCommandReply = socksCommandReplyRW.wrap(writeBuffer,
+                                                                            DataFW.FIELD_OFFSET_PAYLOAD,
+                                                                            writeBuffer.capacity())
+                                                                       .version(5)
+                                                                       .type(t -> t.set(SocksCommandReplyType.SUCCEEDED))
+                                                                       .reserved(0)
+                                                                       .address(a->a.ipv4Address(i->i.set(ipv4Address)))
+                                                                       .port(port)
+                                                                       .build();
+
+            doNetworkData(socksCommandReply.buffer(), socksCommandReply.offset(), socksCommandReply.sizeof());
+        }
+    }
+
+    private final class SocksServerStream
+    {
+        private final SocksServer receiver;
+        private final MessageConsumer application;
+        private long routeId;
+        private long initialId;
+        private long replyId;
+
+        private int initialBudget;
+        private int initialPadding;
+        private int replyBudget;
+        private int replyPadding;
+
+        private long decodeTraceId;
+        private DecoderState decodeState;
+        private int bufferSlot = BufferPool.NO_SLOT;
+        private int bufferSlotOffset;
+
+        SocksServerStream(
```

Comment:
We should probably rename this to `SocksConnectStream` for now, since the logic is currently specific to `CONNECT` command.